workspaces/liferay-ticket-workspace/client-extensions/liferay-ticket-custom-element/src/main.tsx

```diff
@@ -3,26 +3,54 @@
  * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
  */
 
+import {ClayIconSpriteContext} from '@clayui/icon';
 import {createRoot} from 'react-dom/client';
-import {QueryClient, QueryClientProvider} from 'react-query';
+import {QueryClientProvider} from 'react-query';
 
-import TicketApp from './pages/TicketApp';
+import {QueryClientContext} from './context';
+import AllScreens from './pages/AllScreens';
+import TicketsDashboard from './pages/TicketsDashboard';
+import TicketsOverview from './pages/TicketsOverview';
+import {Liferay} from './services/liferay';
+import {ScreenType} from './types';
 
-const queryClient = new QueryClient();
-class TicketWebComponent extends HTMLElement {
-	connectedCallback() {
-		const root = createRoot(this);
+export type LiferayTicketWorkspaceComponentsType = {
+	[key: string]: JSX.Element;
+};
+
+const NoRouteSelected = () => <AllScreens />;
+
+const LiferayTicketWorkspaceComponents: LiferayTicketWorkspaceComponentsType = {
+	dashboard: <TicketsDashboard screenType={ScreenType.STANDALONE} />,
+	overview: <TicketsOverview />,
+};
+
+const DirectToCustomer: React.FC<{defaultScreen: string}> = ({defaultScreen}) =>
+	LiferayTicketWorkspaceComponents[defaultScreen] ?? <NoRouteSelected />;
 
-		root.render(
+const Main: React.FC<{defaultScreen: string}> = ({defaultScreen}) => (
+	<QueryClientContext.Consumer>
+		{(queryClient) => (
```

Comment:
You don't need `QueryClientContext.Consumer` you can delete that component and just instantiate the queryClient in this file and pass it to `QueryClientProvider` in the next line. `QueryClientProvider` is what is providing the queryClient via react context to all of your components.